core/src/main/scala/newtypes.scala

```diff
@@ -82,33 +100,21 @@ object newtypes:
       inline def atLeast(inline bot: A): A      = apply(Math.max(raw(a), raw(bot)))
       inline def atMost(inline top: A): A       = apply(Math.min(raw(a), raw(top)))
 
-  trait OpaqueLong[A](using A =:= Long) extends TotalWrapper[A, Long]
-  trait OpaqueDouble[A](using A =:= Double) extends TotalWrapper[A, Double]:
+  abstract class OpaqueLong[A](using A =:= Long) extends TotalWrapper[A, Long]
+  abstract class OpaqueDouble[A](using A =:= Double) extends TotalWrapper[A, Double]:
     extension (inline a: A) inline def +(inline o: Int): A = apply(raw(a) + o)
-  trait OpaqueFloat[A](using A =:= Float) extends TotalWrapper[A, Float]
+  abstract class OpaqueFloat[A](using A =:= Float) extends TotalWrapper[A, Float]
 
   import scala.concurrent.duration.FiniteDuration
-  trait OpaqueDuration[A](using A =:= FiniteDuration) extends TotalWrapper[A, FiniteDuration]
-
-  abstract class YesNo[A](using ev: Boolean =:= A):
-    val Yes: A = true
```

Comment:
We should keep `Yes` and `No` as we use them occasionally in lila.

Reply:
Will patch